src/qibojit/backends/gpu.py

```diff
@@ -156,13 +156,14 @@ def zero_density_matrix(self, nqubits):
         self.cp.cuda.stream.get_current_stream().synchronize()
         return state.reshape((n, n))
 
-    def identity_density_matrix(self, nqubits):
+    def identity_density_matrix(self, nqubits, normalize: bool = True):
         n = 1 << nqubits
         kernel = self.gates.get(f"initial_state_kernel_{self.kernel_type}")
         state = self.cp.eye(n, dtype=self.dtype)
         kernel((1,), (1,), [state])
```

Comment:
Not directly related to this PR but I believe you can remove the `kernel` calls here
```suggestion
        state = self.cp.eye(n, dtype=self.dtype)
```
All this `kernel` does is `state[0] = 1` but this is not needed if the matrix is already the identity.

If you do so, you should also be able to remove the reshape in the return, since `eye` already has the proper shape.